exercises/clock/.meta/gen.go

```diff
@@ -22,71 +22,60 @@ func main() {
 
 // The JSON structure we expect to be able to umarshal into
 type js struct {
-	Create struct {
-		Description []string
-		Cases       []timeCase
-	}
-	Add struct {
-		Description []string
-		Cases       []addCase
-	}
-	Equal struct {
-		Description []string
-		Cases       []eqCase
+	Cases []struct {
+		Description string
+		Cases       []OneCase
 	}
 }
 
-// Handle the three tests similarly
+type OneCase struct {
+	Description string
+	Property    string
+	Hour        int // "create"/"add" cases
+	Minute      int // "create"/"add" cases
+	Add         int // "add" cases only
 
-type timeCase struct {
-	Description  string
-	Hour, Minute int
-	Expected     string
-}
-type addCase struct {
-	Description       string
-	Hour, Minute, Add int
-	Expected          string
-}
-type eqCase struct {
-	Description    string
-	Clock1, Clock2 struct{ Hour, Minute int }
-	Expected       bool
+	Clock1   struct{ Hour, Minute int } // "equal" cases only
+	Clock2   struct{ Hour, Minute int } // "equal" cases only
+	Expected interface{}                // string or bool
 }
 
+func (c OneCase) IsTimeCase() bool  { return c.Property == "create" }
+func (c OneCase) IsAddCase() bool   { return c.Property == "add" }
+func (c OneCase) IsEqualCase() bool { return c.Property == "equal" }
+
 var tmpl = `package clock
 
-// Source: {{.Ori}}
-{{if .Commit}}// Commit: {{.Commit}}
-{{end}}
+{{.Header}}
 
-{{range .J.Create.Description}}// {{.}}
-{{end}}var timeTests = []struct {
+// Test creating a new clock with an initial time.
```

Comment:
I wondered whether it added much generality, as there are 3 fixed group of test cases.  I suppose if the description of a group changes in canonical, a little extra logic that picked that up can carry the updated description into cases_test.go (instead of my static description comment).

I can work with it a bit.